### ISSUE TITLE
[Accessibility]: Tree page (pre-redesign): Focus lost when navigating from sidebar to page

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35237]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34420

### Problem description

![focus-sidebar-tree](https://user-images.githubusercontent.com/9516420/167640489-cc9c67c0-ffdb-40a6-882c-f5b3e512cdb9.gif)

See how the focus is lost when navigating from the sidebar to the tree page. It takes about 4 presses of <kbd>Tab</kbd> before we finally reach "API Docs"

### Expected behavior

We should expect that:
1. Focus is clear
2. The next tabbed item is the API Docs button (unless there's another appropriate item)

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by felixfbecker** [comment]
Cannot reproduce anymore, the next item is the tab panel now (which however has no focus style, that's a separate issue) and afterwards follows the API docs button directly.
